src/storage/mvcc/txn.rs

```diff
@@ -465,20 +465,20 @@ impl<S: Snapshot> MvccTxn<S> {
     pub fn commit(&mut self, key: Key, commit_ts: TimeStamp) -> Result<bool> {
         let (lock_type, short_value, is_pessimistic_txn) = match self.reader.load_lock(&key)? {
             Some(ref mut lock) if lock.ts == self.start_ts => {
-                // A pessimistic lock cannot be committed.
+                // It's an abnormal routine since pessimistic locks shouldn't be committed in our
+                // transaction model. But a pessimistic lock will be left if the pessimistic
+                // rollback request fails to send and the transaction need not to acquire
+                // this lock again(due to WriteConflict). If the transaction is committed, we
+                // should commit this pessimistic lock too.
                 if lock.lock_type == LockType::Pessimistic {
-                    error!(
-                        "trying to commit a pessimistic lock";
+                    warn!(
+                        "commit a pessimistic lock";
```

Comment:
How about state it clearly that the lock will be removed？

Reply:
The lock is not removed.

Reply:
Sorry. I mean, state it clearly in the log that the lock is committed in Lock type